packages/js/eslint-plugin/src/rules/key-name-casing.ts

```diff
@@ -0,0 +1,161 @@
+import { SourceCode } from "eslint"
+import { JSONProperty } from "jsonc-eslint-parser/lib/parser/ast"
+
+import { OpenverseRule } from "../utils/rule-creator"
+
+export type CasingKind = "camelCaseWithDot" | "snake_case_with_dot"
+
+export const allowedCaseOptions: CasingKind[] = [
+  "camelCaseWithDot",
+  "snake_case_with_dot",
+]
+
+const checkersMap = {
+  camelCaseWithDot: isCamelCase,
+  snake_case_with_dot: isSnakeCase,
+}
+
+/**
+ * Checks whether the given string has symbols.
+ */
+function hasSymbols(str: string) {
+  return /[\u0021-\u0023\u0025-\u002c./\u003a-\u0040\u005b-\u005e`\u007b-\u007d]/u.test(
+    str
+  ) // without " ", "$", "-" and "_"
```

Comment:
Are these ranges inclusive of the bounds? When I check https://symbl.cc/en/unicode-table/, 0025 is the space character. Should the the second range be updated to 0026?

Also: I'm not sure whether the inclusion of `.`, `/` and `` ` `` separate from the unicode ranges is intentional, but if it is, it might be helpful to move them to the start or end of the bracket group so they do not get lost.